### PR TITLE
Fix issue [#1081](https://github.com/briannesbitt/Carbon/issues/1081)

### PR DESCRIPTION
**Pull request by stueliueli**

### CODE REVIEW

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -1227,8 +1227,13 @@ public function formatLocalized($format)
             $format = preg_replace('#(?<!%)((?:%%)*)%e#', '\1%#d', $format); // @codeCoverageIgnore
         }
 
+        $customToStringFormat = self::$toStringFormat;
+        self::resetToStringFormat();
+
         $formatted = strftime($format, strtotime($this));
```
Comment (Glavic) [inline src/Carbon/Carbon.php]: Why don't we just replace 
`$formatted = strftime($format, strtotime($this));` 
with 
`$formatted = strftime($format, $this->getTimestamp());`
?

**Comment by Glavic** [comment]
Other issue I see here is timezone. `strftime()` will return formated date based on user's global timezone...

**Comment by stueliueli** [comment]
@Glavic Yes, but that was already described in the PhpDoc method header: 

> Format the instance with the current locale.  You can set the current locale using setlocale() http://php.net/setlocale.

Would there be any other way this could be done?

**Comment by Glavic** [comment]
What was already described in phpdoc? Missed that.

Only thing that I could think of now is that we would need to set global timezone, format, and then revert global tz...

File: src/Carbon/Carbon.php (hunk 1 above)
Comment (stueliueli) [reply src/Carbon/Carbon.php]: Very good point. Knowing `getTimestamp` makes the use of `strtotime` obsolete.

**Comment by stueliueli** [comment]
Ah, found the reason: getTimestamp returns the timestamp of the current user's timezone. 
According to unitTest `StringsTest::testToLocalizedFormattedTimezonedDateString`, the input is `1975-12-25 14:15:16` which corresponds to the timestamp `188748916`.
If I pass this timestamp to strftime as parameter, I get the output `1975-12-25 09:15:16`, because the carbon-object was created using timezone "Europe/London" and I'm currently in Timezone "Europe/Berlin". 

I don't know if there's any workaround to this @Glavic

File: src/Carbon/Carbon.php (hunk 1 above)
Comment (stueliueli) [reply src/Carbon/Carbon.php]: See https://github.com/briannesbitt/Carbon/pull/1107#issuecomment-366469612

**Comment by Glavic** [comment]
I was talking about the timezone, not locale. You mentioned phpdoc, so I thought you are referring to php.net for locale function :)

Like I said, we could save global timezone, set our timezone, format the date, revert the global timezone, like this: 

```php
$defaultTz = date_default_timezone_get();
date_default_timezone_set($this->getTimezone()->getName());
$formatted = strftime($format, $this->getTimestamp());
date_default_timezone_set($defaultTz);
```

Or we could just let `strtotime()` do the magic and use:
```php
$formatted = strftime($format, strtotime($this->format($this::DEFAULT_TO_STRING_FORMAT)));
```

**Comment by briannesbitt** [comment]
I think just calling `toDateTimeString()` is safer because the `DEFAULT_TO_STRING_FORMAT` could in theory change.  But skipping messing with the `self::resetToStringFormat()` is definitely ideal.

**Comment by stueliueli** [comment]
Oof, always something new to learn. Didn't know `toDateTimeString` but using it seems really the best way. Fixed in Commit https://github.com/briannesbitt/Carbon/pull/1107/commits/c2fdfd7b20190e9efa62168ecadc2facc5c57bdb

**Comment by kylekatarnls** [comment]
I think we have a similar PR, now it has been changed Just have to check that.

**Comment by briannesbitt** [comment]
#1032

**Pull request merged**